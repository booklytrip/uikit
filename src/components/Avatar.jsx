import styled from 'styled-components';
import BaseAvatar from 'react-avatar';

const Avatar = styled(BaseAvatar)`
    ${props => props.spaced === 'left' && `margin-left: 0.5em;`}
    ${props => props.spaced === 'right' && `margin-right: 0.5em;`}
`;

export default Avatar;