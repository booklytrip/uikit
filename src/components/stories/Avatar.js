import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@kadira/react-storybook-decorator-centered';

import Avatar from '../Avatar.jsx';

const styles = {
    sizes: {
        width: '200px',
        textAlign: 'left',
        margin: '0 auto 0 auto',
    },
};

storiesOf('Avatar', module)
    .addDecorator(getStory => {
        return (
            <div className="ui center aligned container">
                {getStory()}
            </div>
        );
    })
    .addDecorator(centered)
    .add('text', () => <Avatar name="John Smith" />)
    .add('text with rounded borders', () => <Avatar name="John Smith" round />);
