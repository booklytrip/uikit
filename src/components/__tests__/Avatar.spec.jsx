import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { ImageAvatar, NameAvatar } from '../Avatar';

describe('ImageAvatar', () => {
    it('renders without crashing', () => {
        mount(<ImageAvatar />);
    });
    
    describe('Snapshots', () => {
        it('renders with specified image', () => {
            const rendered = renderer.create(
                <ImageAvatar image="/avatars/image.png" />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with specified name', () => {
            const rendered = renderer.create(
                <ImageAvatar name="Name" />    
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with different border radius', () => {
            const rendered = renderer.create(
                <ImageAvatar radius={ 10 } />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});

describe('NameAvatar', () => {
    it('renders without crashing', () => {
        mount(<NameAvatar name="User Name" />);
    });
    
    describe('Snapshots', () => {
        it('renders with different number of characters', () => {
            const rendered = renderer.create(
                <NameAvatar name="A B C" maxChars={ 3 } />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with different border radius', () => {
            const rendered = renderer.create(
                <NameAvatar name="A B" radius={ 2 } />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with different text color', () => {
            const rendered = renderer.create(
                <NameAvatar name="A B" textColor="FF0000" />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with different font', () => {
            const rendered = renderer.create(
                <NameAvatar name="A B" fontFamily="Arial" />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with different font weight', () => {
            const rendered = renderer.create(
                <NameAvatar name="A B" fontWeight="400" />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with different font size', () => {
            const rendered = renderer.create(
                <NameAvatar name="A B" fontWeight="12px" />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with different width', () => {
            const rendered = renderer.create(
                <NameAvatar name="A B" width="300" />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with different height', () => {
            const rendered = renderer.create(
                <NameAvatar name="A B" height="300" />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
