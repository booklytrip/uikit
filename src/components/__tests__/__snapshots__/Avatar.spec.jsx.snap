exports[`ImageAvatar Snapshots renders with different border radius 1`] = `
<img
  alt={undefined}
  src={undefined}
  style={
    Object {
      "borderRadius": 10,
    }
  } />
`;

exports[`ImageAvatar Snapshots renders with specified image 1`] = `
<img
  alt={undefined}
  src="/avatars/image.png"
  style={
    Object {
      "borderRadius": 0,
    }
  } />
`;

exports[`ImageAvatar Snapshots renders with specified name 1`] = `
<img
  alt="Name"
  src={undefined}
  style={
    Object {
      "borderRadius": 0,
    }
  } />
`;

exports[`NameAvatar Snapshots renders with different border radius 1`] = `
<svg
  pointerEvents="none"
  style={
    Object {
      "backgroundColor": "#ba83a5",
      "borderRadius": 2,
    }
  }
  viewBox="0 0 undefined undefined">
  <text
    dy="0.35em"
    fill={undefined}
    fontFamily={undefined}
    style={
      Object {
        "fontSize": undefined,
        "fontWeight": undefined,
      }
    }
    textAnchor="middle"
    x="50%"
    y="50%">
    AB
  </text>
</svg>
`;

exports[`NameAvatar Snapshots renders with different font 1`] = `
<svg
  pointerEvents="none"
  style={
    Object {
      "backgroundColor": "#ba83a5",
      "borderRadius": undefined,
    }
  }
  viewBox="0 0 undefined undefined">
  <text
    dy="0.35em"
    fill={undefined}
    fontFamily="Arial"
    style={
      Object {
        "fontSize": undefined,
        "fontWeight": undefined,
      }
    }
    textAnchor="middle"
    x="50%"
    y="50%">
    AB
  </text>
</svg>
`;

exports[`NameAvatar Snapshots renders with different font size 1`] = `
<svg
  pointerEvents="none"
  style={
    Object {
      "backgroundColor": "#ba83a5",
      "borderRadius": undefined,
    }
  }
  viewBox="0 0 undefined undefined">
  <text
    dy="0.35em"
    fill={undefined}
    fontFamily={undefined}
    style={
      Object {
        "fontSize": undefined,
        "fontWeight": "12px",
      }
    }
    textAnchor="middle"
    x="50%"
    y="50%">
    AB
  </text>
</svg>
`;

exports[`NameAvatar Snapshots renders with different font weight 1`] = `
<svg
  pointerEvents="none"
  style={
    Object {
      "backgroundColor": "#ba83a5",
      "borderRadius": undefined,
    }
  }
  viewBox="0 0 undefined undefined">
  <text
    dy="0.35em"
    fill={undefined}
    fontFamily={undefined}
    style={
      Object {
        "fontSize": undefined,
        "fontWeight": "400",
      }
    }
    textAnchor="middle"
    x="50%"
    y="50%">
    AB
  </text>
</svg>
`;

exports[`NameAvatar Snapshots renders with different height 1`] = `
<svg
  pointerEvents="none"
  style={
    Object {
      "backgroundColor": "#ba83a5",
      "borderRadius": undefined,
    }
  }
  viewBox="0 0 undefined 300">
  <text
    dy="0.35em"
    fill={undefined}
    fontFamily={undefined}
    style={
      Object {
        "fontSize": undefined,
        "fontWeight": undefined,
      }
    }
    textAnchor="middle"
    x="50%"
    y="50%">
    AB
  </text>
</svg>
`;

exports[`NameAvatar Snapshots renders with different number of characters 1`] = `
<svg
  pointerEvents="none"
  style={
    Object {
      "backgroundColor": "#816677",
      "borderRadius": undefined,
    }
  }
  viewBox="0 0 undefined undefined">
  <text
    dy="0.35em"
    fill={undefined}
    fontFamily={undefined}
    style={
      Object {
        "fontSize": undefined,
        "fontWeight": undefined,
      }
    }
    textAnchor="middle"
    x="50%"
    y="50%">
    ABC
  </text>
</svg>
`;

exports[`NameAvatar Snapshots renders with different text color 1`] = `
<svg
  pointerEvents="none"
  style={
    Object {
      "backgroundColor": "#ba83a5",
      "borderRadius": undefined,
    }
  }
  viewBox="0 0 undefined undefined">
  <text
    dy="0.35em"
    fill="FF0000"
    fontFamily={undefined}
    style={
      Object {
        "fontSize": undefined,
        "fontWeight": undefined,
      }
    }
    textAnchor="middle"
    x="50%"
    y="50%">
    AB
  </text>
</svg>
`;

exports[`NameAvatar Snapshots renders with different width 1`] = `
<svg
  pointerEvents="none"
  style={
    Object {
      "backgroundColor": "#ba83a5",
      "borderRadius": undefined,
    }
  }
  viewBox="0 0 300 undefined">
  <text
    dy="0.35em"
    fill={undefined}
    fontFamily={undefined}
    style={
      Object {
        "fontSize": undefined,
        "fontWeight": undefined,
      }
    }
    textAnchor="middle"
    x="50%"
    y="50%">
    AB
  </text>
</svg>
`;
